connector/connect/common/src/main/protobuf/spark/connect/types.proto

```diff
@@ -100,6 +100,7 @@ message DataType {
 
   message String {
     uint32 type_variation_reference = 1;
+    optional uint32 collation_id = 2;
```

Comment:
The String type should always have a collation_id so this field should be non optional? The default value is 0.

Reply:
Yes, StringType will always have a collation_id so making it required sounds good.

The thing that worries me here is backwards compatibility. E.g. what happens if other side is on an old version without collation support?

Reply:
According to https://protobuf.dev/programming-guides/proto3/#updating, this change should be both forward and backward-compatible; default value of 0 will be used for this field on newer version when protobuf definitions differ on communication endpoints, on the other side the field would simply be ignored. Removing `optional` per @amaliujia's suggestion.